### ISSUE TITLE
BucketIterator goes into infinite loop

### ISSUE THREAD

**Issue by DeepInEvil** [pytorch/text#231]
When I use the bucketIterator with a tabulardataset object, it is going into infinite loop in training.
Here is the example of the use:
    train = data.TabularDataset(
        path= review_dir + d_review_dir + 'train.tsv', format='tsv', skip_header=True,
        fields=[('text', TEXT),
                ('labels', LABELS)])

    train_iter, dev_iter, test_iter = data.Iterator.splits(
                                (train, val, test), device = -1,
                                batch_sizes=(args.batch_size, len(val), len(test)))

When I loop through the train_iter object with: for batch in train_iter:
it goes into an infinite loop, please fix the error.

**Comment by keitakurita** [comment]
If you do not want the iterator to go into an infinite loop, you need to pass the keyword argument `repeat=False` like

```python
train_iter, dev_iter, test_iter = data.Iterator.splits(
                            (train, val, test), device = -1,
                            repeat=False,
                            batch_sizes=(args.batch_size, len(val), len(test)))
```

**Comment by abhinavarora** [comment]
Closing the issue as we have gotten rid of legacy code and thus this issue is no more relevant, 
#1537
